js_task/script.js

```diff
@@ -0,0 +1,579 @@
+            rating: 4,
+            reviews: [
+                'Delicious sushi, but very delivery.',
+                'Ordered sushi here for a birthday party, everyone was happy.'
+            ]
+        },
+        {
+            id: '19',
+            description: 'Tasty pizza is always a good idea. Especially with 20% discount!',
+            createdAt: new Date('2021-01-27T21:52:44'),
+            link: 'https://domino.by/',
+            vendor: 'Domino Pizza',
+            photoLink: 'https://img.tyt.by/p/05/3/v_statyu_3_dominos221019.jpg',
+            hashTags: [
+                'food', 'pizza', 'fastfood', 'minsk'
+            ],
+            discount: '34%',
+            validUntil: new Date('2021-05-17T00:00:00'),
+            rating: 4,
+            reviews: [
+                'Not bad.'
+            ]
+        },
+        {
+            id: '20',
+            description: 'Develop your mind with our chess school.',
+            createdAt: new Date('2021-03-22T9:21:38'),
+            link: 'https://best-chess-school.by/',
+            vendor: 'Fide',
+            photoLink: 'https://upload.wikimedia.org/wikipedia/commons/d/d9/Opening_chess_position_from_black_side.jpg',
+            hashTags: [
+                'chess', 'mind', 'hobby', 'minsk', 'school'
+            ],
+            discount: '42%',
+            validUntil: new Date('2021-06-20T00:00:00'),
+            rating: 2,
+            reviews: [
+                'Such an awful place with rude teachers.'
+            ]
+        },
+    ];
+
+    function getAds(skip = 0, top = 10, filterConfig = undefined) {
+        if (typeof skip !== 'number' || typeof top !== 'number') {
+            console.log('Error with inputting types!');
+            return;
+        }
+        if (filterConfig) {
+            let returningAds = adList;
+            for (let param in filterConfig) {
+                if (param === 'hashTags') {
+                    for (let i = 0; i < filterConfig.hashTags.length; i++) {
+                        returningAds = returningAds.filter(adItem => adItem.hashTags.includes(filterConfig.hashTags[i]));
+                    }
+                }
+                else if (param === 'dateFrom'){
+                    returningAds = returningAds.filter(adItem => adItem.createdAt >= filterConfig.dateFrom);
+                }
+                else if (param === 'dateTo'){
+                    returningAds = returningAds.filter(adItem => adItem.createdAt < filterConfig.dateTo);
+                }
+                else if (param === 'vendor'){
+                    returningAds = returningAds.filter(adItem => adItem.vendor === filterConfig.vendor);
+                }
+            }
+            returningAds.sort(comparator);
+            return returningAds.slice(skip, skip + top);
+        }
+        else {
+            let returningAds = adList.slice(skip, skip + top);
+            returningAds.sort(comparator);
+            return returningAds;
+        }
```

Comment:
Looks completely the same as other students' method (even string you log to console), so I'll decrease score for each of you who used this code.